in_toto/resolver/_resolver.py

```diff
@@ -205,3 +205,71 @@ def hash_artifacts(self, uris):
             os.chdir(original_cwd)
 
         return hashes
+
+
+class OSTreeResolver(Resolver):
+    """Resolver for OSTree repositories. """
+
+    SCHEME = "ostree"
+
+    # defining this separately as it should be pinned to OSTree's default
+    # rather than in-toto's
+    _HASH_ALGORITHM = "sha256"
+
+    def __init__(self, base_path=None):
+        self._base_path = base_path
+
+    def _strip_scheme_prefix(self, path):
+        """Helper to strip OSTree resolver scheme prefix from path."""
+
+        return path[len(self.SCHEME + ":") :]
+
+    def _add_scheme_prefix(self, path):
+        """Helper to add scheme back after recording the hash."""
+
+        return f"{self.SCHEME}:{path}"
+
+    def _hash(self, path):
+        """Helper to hash OSTree commits."""
+
+        ref_path = os.path.join("refs", "heads", path)
+        if not os.path.exists(ref_path):
+            raise OSTreeRefNotFound(f"OSTree ref '{path}' not found")
```

Comment:
The check here doesn't guarantee that there won't be a `FileNotFoundError` below, or a different error from open (missing permissions etc.) 

Do you anticipate users to handle `OSTreeRefNotFound`? If not, I'd just use the builtins.

Reply:
Removed the exception altogether.